tools/sync.js

```diff
@@ -0,0 +1,214 @@
+          var lastSync = blockData.number;
+          updateLastSynced(config, lastSync);
+        }else{
+          console.log('Found existing last Sync');
+          writeBlockToDB(config, blockData);
+          writeTransactionsToDB(config, blockData);
+          var lastSync = config.lastSynced - 1;
+          updateLastSynced(config, lastSync);
+        }
+      }
+    });
+  }else{
+    console.log('Error: Web3 connection time out trying to get block ' + blockHashOrNumber + ' retrying connection now');
+    return;
+  }
+};
+var writeBlockToDB = function(config, blockData) {
+  return new Block(blockData).save( function( err, block, count ){
+    if ( typeof err !== 'undefined' && err ) {
+        if (err.code == 11000) {
+            if(!('quiet' in config && config.quiet === true)) {
+              console.log('Skip: Duplicate key ' +   blockData.number.toString() + ': ' + err);
+            }
+        } else {
+           console.log('Error: Aborted due to error on ' + 'block number ' + blockData.number.toString() + ': ' +  err);
+           process.exit(9);
+       }
+    } else {
+        if(!('quiet' in config && config.quiet === true)) {
+            console.log('DB successfully written for block number ' + blockData.number.toString());
+        }
+    }
+  });
+}
+/**
+    Break transactions out of blocks and write to DB
+**/
+var writeTransactionsToDB = function(config, blockData) {
+  var bulkOps = [];
+  if (blockData.transactions.length > 0) {
+      for (d in blockData.transactions) {
+          var txData = blockData.transactions[d];
+          txData.timestamp = blockData.timestamp;
+          txData.value = etherUnits.toEther(new BigNumber(txData.value), 'wei');
+          bulkOps.push(txData);
+      }
+      Transaction.collection.insert(bulkOps, function( err, tx ){
+          if ( typeof err !== 'undefined' && err ) {
+              if (err.code == 11000) {
+                  console.log('Skip: Duplicate key ' +
+                  err);
+              } else {
+                 console.log('Error: Aborted due to error: ' +
+                      err);
+                 process.exit(9);
+             }
+          } else if(!('quiet' in config && config.quiet === true)) {
+              console.log(blockData.transactions.length.toString() + ' transactions recorded for Block# ' + blockData.number.toString());
+          }
+      });
+  }
+}
+var checkBlockDBExistsThenWrite = function(config, blockData) {
+  Block.find({number: blockData.number}, function (err, b) {
+      if (!b.length){
+          writeBlockToDB(config, blockData);
+          writeTransactionsToDB(config, blockData);
+      }else if(!('quiet' in config && config.quiet === true)) {
+          console.log('Block number: ' + blockData.number.toString() + ' already exists in DB.');
+          listenBlocks(config);
+      }
+  });
+};
+var updatedEndBlock = function(config,lastBlock){
+  var configFile = '../conf.json';
```

Comment:
One here is `../conf.json` other is `conf.json`

Also I'm not familiar with `fs.writeFile()` but I'd think that since you already have the `file` handle you'd use that to handle writing `conf.json` everywhere else no?

As mentioned though I don't think writing to `conf.json` for every block is worthwhile as there's lots of writes and lots of risk for user error. Up to you though.

Reply:
I agree with writing to the config file to keep memory. Future version will have a better way to handle this.